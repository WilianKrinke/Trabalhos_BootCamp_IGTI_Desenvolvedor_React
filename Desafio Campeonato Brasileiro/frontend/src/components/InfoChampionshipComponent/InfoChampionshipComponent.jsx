import React from 'react';
import { number } from 'prop-types';

const InfoChampionshipComponent = ({anoDoCampeonato, rodada}) => {
    return (
        <>
            <h3>Campeonato Brasileiro de {anoDoCampeonato}</h3>
            <h3>{rodada}Âª Rodada</h3>
        </>
    );
}

InfoChampionshipComponent.propTypes = {
    anoDoCampeonato: number.isRequired,
    rodada: number.isRequired
}


export default InfoChampionshipComponent;
