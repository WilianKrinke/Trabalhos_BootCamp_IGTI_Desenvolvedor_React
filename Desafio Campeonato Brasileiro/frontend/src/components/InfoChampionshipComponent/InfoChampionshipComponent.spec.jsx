import React from 'react'
import InfoChampionshipComponent from './InfoChampionshipComponent'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import {render, screen} from '@testing-library/react'

describe('InfoChampionshipComponent test component', () => {
    it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<InfoChampionshipComponent anoDoCampeonato={2006} rodada={38}/>,div)
    });

    it('should render with texts', () => {
        render(<InfoChampionshipComponent anoDoCampeonato={2006} rodada={38}/>)

        const h3ElementAno = screen.getByText(/Campeonato Brasileiro de 2006/i)
        expect(h3ElementAno).toBeInTheDocument();
        
        const h3ElementRodada = screen.getByText(/38Âª Rodada/i)
        expect(h3ElementRodada).toBeInTheDocument();
    });

    it('matches snapshot 1', () => {
        const tree = renderer.create(
            <InfoChampionshipComponent anoDoCampeonato={2006} rodada={38}/>
        ).toJSON()
        expect(tree).toMatchSnapshot();
    });
    
});



