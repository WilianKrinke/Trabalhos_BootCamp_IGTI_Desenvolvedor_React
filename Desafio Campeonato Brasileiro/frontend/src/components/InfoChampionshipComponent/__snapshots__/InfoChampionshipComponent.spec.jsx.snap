// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoChampionshipComponent test component matches snapshot 1 1`] = `
Array [
  <h3>
    Campeonato Brasileiro de 
    2006
  </h3>,
  <h3>
    38
    Âª Rodada
  </h3>,
]
`;
